app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt

```diff
@@ -30,6 +30,10 @@ class SearchInteractor(
     private val store: SearchStore
 ) : AwesomeBarInteractor, ToolbarInteractor {
 
+    data class UserTypingCheck(var ranOnTextChanged: Boolean, var userHasTyped: Boolean)
+
+    private val startedTyping = UserTypingCheck(false, !store.state.showShortcutEnginePicker)
```

Comment:
One final nit: can we name this something like `typingCheck` or `userTypingCheck` since that more closely matches the data class you've created :)